apps/web/pages/auth/setup/index.tsx

```diff
@@ -38,8 +41,22 @@ export default function Setup(props: inferSSRProps<typeof getServerSideProps>) {
   );
 }
 
-export const getServerSideProps = async () => {
+export const getServerSideProps = async (context: GetServerSidePropsContext) => {
   const userCount = await prisma.user.count();
+  const { req } = context;
+  const session = await getSession({ req });
+
+  console.log({ role: session?.user.role });
+
+  if (session?.user.role && session?.user.role !== UserPermissionRole.ADMIN) {
+    return {
+      redirect: {
+        destination: `/400`,
```

Comment:
403 or 404 seems better option to me.
403 - Clearly tells that you are not authorized
404 - We want to hide that the user is accessing a valid route

Reply:
@hariombalhara I think we only have error 400 and 500 at the moment

Reply:
@leog we don't have a 400 error page tho

Reply:
My bad! 😅